### PR TITLE
fix: Display offset for filenames with spaces

### PR DESCRIPTION
**Pull request by guillaumeboehm**
Fix for #547

I did the fix somewhat quickly so I might have broken something else.
If I missed something please give me a little bit of context and I'll fix it!

### CODE REVIEW

**Review by PThorpe92 — COMMENTED** [review]
Thanks for the PR :)

Can we handle the case of multiple spaces? Also shadowing a variable `offset` is probably how I would do it as opposed to a mutable one.


@cafkafk I think perhaps we need some nix tests with 'spaces in filenames'.json

I'll generate some later and test this PR against it.

File: src/output/grid.rs [hunk 1]
```diff
@@ -56,21 +56,28 @@ impl<'a> Render<'a> {
                     0
                 };
 
+            let mut additionnal_chars = 0;
```
Comment (PThorpe92) [inline src/output/grid.rs:59]: Can we do something like this?
```Rust
let offset = if file.name.contains(' ') { 2 } else { 0 }

Comment (guillaumeboehm) [reply src/output/grid.rs:59]: Yeah ok works for me if you don't feel like we'll need the offset for something else. I'll fix it quickly when I get home.

Comment (PThorpe92) [reply src/output/grid.rs:59]: Actually now that I look at it. Maybe we should do 
```Rust
let offset = if file.name.contains(' ') { 3 } else { 1 }
// .....
ShowIcons::Automatic(spacing)) => filename.bare_width() + classification_width + offset + (spacing as usize)
// etc.. 
```
And then we can remove the hardcoded ` + 1` value

**Comment by guillaumeboehm** [comment]
Renamed to "offset" and fixed the display for the hyperlink option as well.

File: src/output/grid.rs (hunk 1 above)
Comment (guillaumeboehm) [reply src/output/grid.rs]: Well the +2 is also needed for the hyperlink only case

**Comment by guillaumeboehm** [comment]
Felt like a bit more clarity wouldn't hurt.

**Comment by PThorpe92** [comment]
Oh my bad.. totally didn't mean to close this. I was just fixing the rustfmt issue in the build. for some reason it closed the PR. 

Definitely open a new one lol my fault :heart:

**Comment by PThorpe92** [comment]
reopening

**Comment by guillaumeboehm** [comment]
> reopening..
> 
> EDIT: yeah it's not letting me re-open this. Something similar happened for a PR I had in here a while ago. It will let me edit it on github, but if I make changes from my local machine and push to your fork, it auto-closes the PR and wont let me open it again. My fault

Wait I'm so confused xD should I open a new PR ?

**Comment by PThorpe92** [comment]
Lol yeah at first i thought i could re-open this one. You did the right thing

**Pull request closed without merging**